src/node/internal/internal_zlib.ts

```diff
@@ -136,89 +185,79 @@ export function inflate(
+      });
+    });
+
+    return;
+  }
+
+  processChunkCaptureError(new CLASS_BY_MODE[mode](options), data, callback);
+}
+
+export function inflate(
+  data: ArrayBufferView | string,
+  optionsOrCallback: ZlibOptions | CompressCallback,
+  callbackOrUndefined?: CompressCallback
+): void {
+  zlibImpl(CONST_INFLATE, data, optionsOrCallback, callbackOrUndefined);
 }
 
 export function unzip(
   data: ArrayBufferView | string,
   optionsOrCallback: ZlibOptions | CompressCallback,
   callbackOrUndefined?: CompressCallback
 ): void {
-  const [options, callback] = normalizeArgs(
-    optionsOrCallback,
-    callbackOrUndefined
-  );
-  zlibUtil.zlib(data, options, zlibUtil.CONST_UNZIP, wrapCallback(callback));
+  zlibImpl(CONST_UNZIP, data, optionsOrCallback, callbackOrUndefined);
 }
 
 export function inflateRaw(
   data: ArrayBufferView | string,
   optionsOrCallback: ZlibOptions | CompressCallback,
   callbackOrUndefined?: CompressCallback
 ): void {
-  const [options, callback] = normalizeArgs(
-    optionsOrCallback,
-    callbackOrUndefined
-  );
-  zlibUtil.zlib(
-    data,
-    options,
-    zlibUtil.CONST_INFLATERAW,
-    wrapCallback(callback)
-  );
+  zlibImpl(CONST_INFLATERAW, data, optionsOrCallback, callbackOrUndefined);
 }
 
 export function gunzip(
   data: ArrayBufferView | string,
   optionsOrCallback: ZlibOptions | CompressCallback,
   callbackOrUndefined?: CompressCallback
 ): void {
-  const [options, callback] = normalizeArgs(
-    optionsOrCallback,
-    callbackOrUndefined
-  );
-  zlibUtil.zlib(data, options, zlibUtil.CONST_GUNZIP, wrapCallback(callback));
+  zlibImpl(CONST_GUNZIP, data, optionsOrCallback, callbackOrUndefined);
 }
 
 export function deflate(
   data: ArrayBufferView | string,
   optionsOrCallback: ZlibOptions | CompressCallback,
   callbackOrUndefined?: CompressCallback
 ): void {
-  const [options, callback] = normalizeArgs(
-    optionsOrCallback,
-    callbackOrUndefined
-  );
-  zlibUtil.zlib(data, options, zlibUtil.CONST_DEFLATE, wrapCallback(callback));
+  zlibImpl(CONST_DEFLATE, data, optionsOrCallback, callbackOrUndefined);
 }
 
 export function deflateRaw(
   data: ArrayBufferView | string,
   optionsOrCallback: ZlibOptions | CompressCallback,
   callbackOrUndefined?: CompressCallback
 ): void {
-  const [options, callback] = normalizeArgs(
-    optionsOrCallback,
-    callbackOrUndefined
-  );
-  zlibUtil.zlib(
-    data,
-    options,
-    zlibUtil.CONST_DEFLATERAW,
-    wrapCallback(callback)
-  );
+  zlibImpl(CONST_DEFLATERAW, data, optionsOrCallback, callbackOrUndefined);
 }
 
 export function gzip(
   data: ArrayBufferView | string,
   optionsOrCallback: ZlibOptions | CompressCallback,
   callbackOrUndefined?: CompressCallback
 ): void {
-  const [options, callback] = normalizeArgs(
-    optionsOrCallback,
-    callbackOrUndefined
-  );
-  zlibUtil.zlib(data, options, zlibUtil.CONST_GZIP, wrapCallback(callback));
+  zlibImpl(CONST_GZIP, data, optionsOrCallback, callbackOrUndefined);
```

Comment:
Nit: Can we change the variables names? wdyt?

```suggestion
  zlibImpl(CONST_GZIP, data, options, callback);
```